File: front-end/src/app/model/OffreDeStage.ts
export interface OffreDeStage {
  id: number;
  poste: string;
  entreprise: string;
  description: string;
  dateDebut: Date;
  dateFin: Date;
  technologie: string;
  validated: boolean;
}

export interface OffreDeStageDetails {
  id: number;
  poste: string;
  entreprise: string;
  description: string;
  dateDebut: Date;
  dateFin: Date;
  technologie: string;
  adminLogin: string;
  adminPassword: string;
  validated: boolean;
}

File: back-end/src/main/java/ma/enset/gestionStage/entities/OffreDeStage.java
package ma.enset.gestionStage.entities;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.util.Date;

@Entity
@Table(name = "offres_de_stage")
@Data @Getter @Setter
public class OffreDeStage {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;
    private String poste;
    private String entreprise;
    private String description;
    private Date dateDebut;
    private Date dateFin;
    private String technologie;
    private Boolean validated;
}